lib/open-uri.rb

```diff
@@ -27,7 +27,18 @@ class << self
   # We can accept URIs and strings that begin with http://, https:// and
   # ftp://. In these cases, the opened file object is extended by OpenURI::Meta.
   def open(name, *rest, &block) # :doc:
-    if name.respond_to?(:open)
+    if name.respond_to?(:to_io)
```

Comment:
Thanks for that! I changed the fix back to checking the arity of the `open` method instead. What do you think about that?